trillian/ctfe/configpb/config.proto

```diff
@@ -132,6 +133,34 @@ message LogConfig {
   // A list of X.509 extension OIDs, in dotted string form (e.g. "2.3.4.5")
   // which should cause submissions to be rejected.
   repeated string reject_extensions = 18;
+
+  // CTFE storage connection string in the following format in general:
+  // driver://[username[:password]@][protocol[(host[:port])]][/[schema|database][?options]]
+  //
+  // MySQL/MariaDB:
+  // mysql://[username[:password]@][protocol[(address)]]/dbname[?param1=value1&...&paramN=valueN]
+  //
+  // This is required when the non-leaf certificate chain storage backend is
+  // CTFE.
+  // 
+  // Warning: CT log operators are advised not to re-use the same connection 
+  // string across multiple LogConfigs due to the log lifecycle.
+  string ctfe_storage_connection_string = 20;
+
+  // An optional list of storage backends for non-leaf certificate chain in 
+  // ExtraData. Log operators may want to use both Trillian GRPC and CTFE as  
+  // the storage backends to ensure the data availability at the beginning. 
+  // By default, the storage backend is Trillian GRPC. To use CTFE as the  
+  // storage backend, the CTFE storage connection string needs to be specified.
+  enum NonLeafCertificateChainStorageBackend {
+    NON_LEAF_CERTIFICATE_CHAIN_STORAGE_BACKEND_TRILLIAN_GRPC = 0;
+    NON_LEAF_CERTIFICATE_CHAIN_STORAGE_BACKEND_CTFE = 1;
+  }
+  repeated NonLeafCertificateChainStorageBackend extra_data_non_leaf_certificate_chain_storage_backends = 21;
```

Comment:
This being a repeated field seems more complicated than necessary. My recollection was that this would be set during log init, and then left alone. Allowing both seems contrary to that, and suggest that your implementation will include some ability to hot-switch between the two storage mechanisms?

Reply:
I do agree this is a bit complicated. Allowing both storage backends also add the code complexity but it helps ensure the data is stored in both data storage if log operator is not confident with the data availability of the new storage.

The implementation will try to retrieve the chain from one and fallback to another if it couldn't be found.

Reply:
I think the simplicity of a single location to store chain data feels like the better path at the moment:
- there's no ambiguity around chain data location, it's either in the ctfe store, or ExtraData for the lifetime of the log.
- the risk of changing the location by editing the ctfe config is a pretty dangerous sharp edge, there's no way to catch & warn of a mistake that would be terminal for the log, rollbacks are potentially a bit more _exciting_, etc.

If the driver for the extra complexity is primarily to increase confidence in correctness, then an alternative solution would be to bake it on a test log for a good period of time, with the CT hammer or similar continuously verifying that everything is working as intended?

Reply:
Thanks @mhutchinson and @AlCutter for the input. The original design to have both storage running does not really fit into the easy rollback and data resilience objective since the original type can not be used for marshal and unmarshal. Storing a new type in the `ExtraData` field means that data migration is required for rollback.

@phbnf How does this sound to you if this is simplified to have a single storage backend? The risk can be mitigated by integration tests and test logs verification.

Reply:
> I think the simplicity of a single location to store chain data feels like the better path at the moment
+1000 for this.

> @phbnf How does this sound to you if this is simplified to have a single storage backend?
SGTM - and crucially, it should *not* be possible to change which backend storage is used through the lifetime of a log.

Reply:
Thanks all for the comments. Removed `repeated` from `extra_data_issuance_chain_storage_backend`.